packages/integration-tests/src/test-project-setup/conversation_handler_project.ts

```diff
@@ -1189,13 +1189,14 @@ class ConversationHandlerTestProject extends TestProjectBase {
   };
 
   /**
-   * Bedrock sometimes produces empty response or half backed response.
+   * Bedrock sometimes produces half-baked response or we run into throttling.
```

Comment:
are we not exposing the `ThrottlingException` back to the user? It seems that we are swallowing the error and producing empty results.

Reply:
> are we not exposing the ThrottlingException back to the user?

In real e2e use case client libraries would read `errors` from GQL response and handle that.
This test's goal is to test GQL -> lambda integration, we do mimic some GQL calls, but not handle them exactly like libraries. (fwiw this we have this case to ensure that errors are propagating https://github.com/aws-amplify/amplify-backend/blob/11b40c761fe161e974f9991bc3faf3702dd99726/packages/integration-tests/src/test-project-setup/conversation_handler_project.ts#L1061 )

Reply:
> In real e2e use case client libraries would read errors from GQL response and handle that.

Btw. I was thinking about changing https://github.com/aws-amplify/amplify-backend/blob/11b40c761fe161e974f9991bc3faf3702dd99726/packages/integration-tests/src/test-project-setup/conversation_handler_project.ts#L845-L850 and maybe https://github.com/aws-amplify/amplify-backend/blob/11b40c761fe161e974f9991bc3faf3702dd99726/packages/integration-tests/src/test-project-setup/conversation_handler_project.ts#L932-L937 .
to throw and handle that in caller code. But it wouldn't be changing the test and retry logic much, so I refrained from doing so.

Reply:
> so I refrained from doing so.

Originally I was thinking about this when adding the `assertDefaultConversationHandlerCanPropagateError` test case. Where I thought that asserting GQL error response will be somewhat obscured if we translate GQL error to JS Error.